src/changeRemoteModal.jsx

```diff
@@ -0,0 +1,261 @@
+                          isChecked={newRepoTrusted}
+                          onChange={(checked, ev) => {
+                              setNewRepoTrusted(checked);
+                          }} />
+            </FormGroup>
+            <ActionGroup>
+                <Button id="add-remote-btn" onClick={() => onAddRemote()} variant="primary">{_("Add")}</Button>
+                <Button onClick={() => setAddNewRepoDialogOpen(false)} variant="link">{_("Cancel")}</Button>
+            </ActionGroup>
+        </Form>
+    );
+};
+AddNewRepoForm.propTypes = {
+    refreshRemotes: PropTypes.func.isRequired,
+    setAddNewRepoDialogOpen: PropTypes.func.isRequired,
+};
+
+const EditRemoteForm = ({ remoteSettings, setEditRepoDialogOpen, refreshRemotes }) => {
+    const [addAnotherKey, setAddAnotherKey] = useState(false);
+    const [key, setKey] = useState('');
+    const [isTrusted, setIsTrusted] = useState(remoteSettings['gpg-verify'] !== 'false');
+    const [error, setError] = useState('');
+
+    const onUpdate = () => {
+        const promises = [];
+        if (key)
+            promises.push(remotes.importGPGKey(remoteSettings.name, key));
+        promises.push(remotes.updateRemoteSettings(remoteSettings.name, { "gpg-verify": isTrusted }));
+
+        Promise.all(promises).then(() => setEditRepoDialogOpen(false), ex => setError(ex.message));
+    };
+    const onDelete = () => {
+        remotes.deleteRemote(remoteSettings.name)
+                .then(() => refreshRemotes())
+                .then(setEditRepoDialogOpen(false), ex => setError(ex.message));
+    };
+
+    return (
+        <Form isHorizontal>
+            {error && <Alert variant="danger" isInline
+                             action={<AlertActionCloseButton onClose={() => this.setState({ error: undefined })} />}
+                             title={error} />}
+            <Title headingLevel="h3" size="l">
+                {remoteSettings.name}
+            </Title>
+            <FormGroup label={_("URL")}
+                fieldId="edit-remote-url">
+                <TextInput id="edit-remote-url"
+                           value={remoteSettings.url}
+                           readOnly
+                           type="text" />
+            </FormGroup>
+            <FormGroup fieldId="edit-remote-trusted">
+                <Checkbox label={_("Use trusted GPG key")}
+                          id="gpg-verify"
+                          isChecked={isTrusted}
+                          onChange={(checked, ev) => {
+                              setIsTrusted(!isTrusted);
+                          }} />
+            </FormGroup>
+            <FormGroup fieldId="add-another-key">
+                {!addAnotherKey ? <Button isInline variant="secondary" id='add-another-key' onClick={() => setAddAnotherKey(true)}>{_("Add Another Key")}</Button>
+                 : <TextArea id='gpg-data'
+                             placeholder="Begins with '-----BEGIN GPG PUBLIC KEY BLOCK-----'"
```

Comment:
This on the other hand needs *more* i18n; ideally without the `---BEGIN` stuff. Perhaps `_("Begins with '$0'`")` and cockpit.format?